### PR TITLE
NXP backend: Enable `constant_pad_nd` with new Neutron flow.

### PR DESCRIPTION
**Pull request by MartinPavella**
### Summary
Enable `constant_pad_nd` with new Neutron flow in NXP backend. Reflect the new Neutron requirements and verify correct function.

### Test plan
Unit tests provided.

cc @robert-kalmar @JakeStevens @digantdesai @rascani

### CODE REVIEW

**Comment by MartinPavella** [comment]
@novak-vaclav as per usual.

File: backends/nxp/backend/ir/converter/node_converters/ops_converters/constant_pad_nd_converter.py [hunk 1]
```diff
@@ -41,19 +41,25 @@ def _is_supported_on_target(
...
-        else:
-            # Dim `-1` will end up being the channels. It is padded by paddings[:2].
-            if len(paddings) > 0 and paddings[:2] != [0, 0]:
-                # Attempt to Pad channels dimension -> currently not supported
-                return False
+        if custom_delegation_options.use_new_flow_neutron_c:
+            # Requirements specified by the new Neutron flow documentation.
 
-        return True
+            return True  # There are no requirements.
```
Comment (novak-vaclav) [inline backends/nxp/backend/ir/converter/node_converters/ops_converters/constant_pad_nd_converter.py]: The check for io quantization is missing

File: backends/nxp/tests/ir/converter/node_converter/test_constant_pad_nd_converter.py [hunk 2]
```diff
@@ -4,9 +4,15 @@
 # LICENSE file in the root directory of this source tree.
 
 import numpy as np
+
+# noinspection PyUnusedImports
```
Comment (novak-vaclav) [inline backends/nxp/tests/ir/converter/node_converter/test_constant_pad_nd_converter.py:8]: nit: why is this necessary? pytest module is used and linter should not complain

File: backends/nxp/tests/ir/converter/node_converter/test_constant_pad_nd_converter.py [hunk 3]
```diff
@@ -216,6 +217,118 @@ def test_constant_pad_nd__delegation__channels_first__unsupported_padding(use_qa
     ).exported_program()
 
     # Make sure the `pad` was NOT delegated.
-    assert graph_contains_any_of_ops(
-        exec_program.graph, [exir_ops.edge.aten.constant_pad_nd.default]
+    assert graph_contains_any_of_ops(exec_program.graph, [ConstantPadND])
+
+
+class TestConstantPadNDNewNeutronFlow:
```
Comment (novak-vaclav) [inline backends/nxp/tests/ir/converter/node_converter/test_constant_pad_nd_converter.py:221]: nit: I would add comment somewhere that the padding tuples go from the last dimension to the first. It took me a while to understand it and a simple comment or hands-on example would be nice. 
Something like
```
Order of inserted padding is backwards, ie:
p2d = (1, 1, 2, 2) # pad the last dim on both sides and second to last dim on both sides
```

**Review by novak-vaclav — CHANGES_REQUESTED** [review]
Great work! 👍 only a few pointers

File: backends/nxp/tests/ir/converter/node_converter/test_constant_pad_nd_converter.py [hunk 4]
```diff
@@ -216,6 +217,118 @@ def test_constant_pad_nd__delegation__channels_first__unsupported_padding(use_qa
...
+            pytest.param((2, 4), tuple(range(4)), id="2D, padding N, H"),
+            pytest.param((2, 4, 6), tuple(range(2)), id="3D, padding H"),
+            pytest.param((2, 4, 6), tuple(range(4)), id="3D, padding C, H"),
+            pytest.param((2, 4, 6, 8), tuple(range(2)), id="4D, padding W"),
+            pytest.param((2, 4, 6, 8), tuple(range(4)), id="4D, padding H, W"),
+            pytest.param((1, 2, 3, 4, 5), tuple(range(2)), id="5D, padding D"),
+            pytest.param((1, 2, 3, 4, 5), tuple(range(4)), id="5D, padding W, D"),
+        ],
+    )
+    def test__basic_nsys_inference(self, mocker, input_shape, paddings, use_qat):
```
Comment (roman-janik-nxp) [inline backends/nxp/tests/ir/converter/node_converter/test_constant_pad_nd_converter.py:268]: Don't forget QAT variant.

File: backends/nxp/tests/model_output_comparator.py [hunk 5]
```diff
@@ -91,7 +91,13 @@ def compare_sample(self, sample_dir, cpu_output_tensors, npu_output_tensors):
...
                 cpu_tensor
             ), "Output tensor contains only zeros. This is suspicious."
-            assert np.allclose(cpu_tensor, npu_tensor, atol=self.atol)
+            all_close = np.allclose(cpu_tensor, npu_tensor, atol=self.atol)
+            if not all_close:
+                max_diff = np.abs(cpu_tensor - npu_tensor).max()
+                print(
+                    f"NPU output doesn't match reference. Maximum absolute difference: {max_diff}"
+                )
+            assert all_close
```
Comment (roman-janik-nxp) [inline backends/nxp/tests/model_output_comparator.py:100]: Nice.

File: backends/nxp/tests/ir/converter/node_converter/test_constant_pad_nd_converter.py (hunk 2 above)
Comment (MartinPavella) [reply backends/nxp/tests/ir/converter/node_converter/test_constant_pad_nd_converter.py:8]: My IDE (CLion) doesn't recognize usages of `pytest` in decorators. So when I optimize the imports (`ctrl + O`), it removes the `import pytest`.
Apparently this is a known issue so I added the directive comment.

File: backends/nxp/tests/ir/converter/node_converter/test_constant_pad_nd_converter.py (hunk 3 above)
Comment (MartinPavella) [reply backends/nxp/tests/ir/converter/node_converter/test_constant_pad_nd_converter.py:221]: There is a comment in the `ConstantPadNDConverter`:
> The PyTorch padding is added to the individual dimensions from the back (slightly confusing), see:
             https://pytorch.org/docs/stable/generated/torch.nn.functional.pad.html#torch.nn.functional.pad

Paddings are defined tens of times in the testing module, sometimes with inline comments explaining their purpose. Adding a comment to every line would therefore be really messy.
Adding 1 comment on top level would sort of duplicate the one in `ConstantPadNDConverter`. But that's fine I guess. I will add one.

File: backends/nxp/tests/ir/converter/node_converter/test_constant_pad_nd_converter.py (hunk 4 above)
Comment (MartinPavella) [reply backends/nxp/tests/ir/converter/node_converter/test_constant_pad_nd_converter.py:268]: The test already uses the `use_qat` fixture.

File: backends/nxp/backend/ir/converter/node_converters/ops_converters/constant_pad_nd_converter.py (hunk 1 above)
Comment (MartinPavella) [reply backends/nxp/backend/ir/converter/node_converters/ops_converters/constant_pad_nd_converter.py]: Oops, I forgot that. Nice catch!

**Review by novak-vaclav — APPROVED** [review]
I'm still getting errors that I am unable to resolve this thread, but everything looks good to me 😊

Image

**Review by roman-janik-nxp — APPROVED** [review]

**Review by robert-kalmar — APPROVED** [review]

**Comment by MartinPavella** [comment]
The skipped checks seem unrelated and the [internal CI](https://bamboo3.sw.nxp.com/browse/MLTECE-EXIGH156-5) is passing. Merging.
